### PR TITLE
#164829189 update django and all its dependencies

### PR DESCRIPTION
**Pull request by Sm0k3s**
#### Description
Updates django from version 1.10 to the latest version, which prompted to update some of the code to avoid regression. I also updated the other dependencies to their current versions
#### Type of change
Updated the codebase to the latest conventions
#### Checklist:
The updated links still work
#### Pivotal Tracker stories
[#164829189](https://www.pivotaltracker.com/n/projects/2321088/stories/164829189)

### CODE REVIEW

File: authors/apps/authentication/urls.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
     LoginAPIView, RegistrationAPIView, UserRetrieveUpdateAPIView
 )
 
+app_name='authentication'
+
 urlpatterns = [
-    url(r'^user/?$', UserRetrieveUpdateAPIView.as_view()),
-    url(r'^users/?$', RegistrationAPIView.as_view()),
-    url(r'^users/login/?$', LoginAPIView.as_view()),
+    path('user/', UserRetrieveUpdateAPIView.as_view()),
```
Comment (ValerieRono) [inline authors/apps/authentication/urls.py]: add names to our paths so that we can user `reverse()`  in our tests
i.e. register, login, e.t.c

**Review by bl4ck4ndbr0wn — APPROVED** [review]
Good job.

File: authors/apps/authentication/urls.py (hunk 1 above)
Comment (Sm0k3s) [reply authors/apps/authentication/urls.py]: I have made those changes

Comment (ValerieRono) [reply authors/apps/authentication/urls.py]: great

**Review by ValerieRono — APPROVED** [review]

**Pull request merged**